### PR TITLE
CATROID-641 Refresh landing page after download is finished

### PR DESCRIPTION
**Pull request by juliajulie95**
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/config/lint-baseline.xml [hunk 1]
```diff
@@ -173,7 +173,7 @@
         errorLine2="         ~~~~~~">
         <location
             file="src\main\java\org\catrobat\catroid\ui\fragment\FormulaEditorFragment.java"
-            line="100"
+            line="101"
```
Comment (ndrnour) [inline catroid/config/lint-baseline.xml:176]: please revert the changes to this file, see my next comment to know how to get rid of `EmptyFunctionBlock `lint warning

File: catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @Suppress("EmptyFunctionBlock")
+    override fun onDownloadStarted(string: String) {
+    }
+
+    @Suppress("EmptyFunctionBlock")
+    override fun onDownloadProgress(progress: Int, url: String) {
+    }
+
+    @Suppress("EmptyFunctionBlock")
+    override fun onUserCanceledDownload(url: String) {
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt:47]: please change to `override fun onUserCanceledDownload(url: String) = Unit` to get rid of the lint warning

File: catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    override fun onDownloadFinished(catrobatProgramName: String, url: String) {
+        fragment?.get()?.setAndLoadCurrentProject() }
+
+    @Suppress("EmptyFunctionBlock")
+    override fun onDownloadStarted(string: String) {
+    }
+
+    @Suppress("EmptyFunctionBlock")
+    override fun onDownloadProgress(progress: Int, url: String) {
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt:43]: `override fun onDownloadProgress(progress: Int, url: String) = Unit
`

File: catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    private var fragment: WeakReference<MainMenuFragment>? = null
+    fun setFragment(fragment: MainMenuFragment) {
+        this.fragment = WeakReference<MainMenuFragment>(fragment)
+    }
+
+    override fun onDownloadFinished(catrobatProgramName: String, url: String) {
+        fragment?.get()?.setAndLoadCurrentProject() }
+
+    @Suppress("EmptyFunctionBlock")
+    override fun onDownloadStarted(string: String) {
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt:39]: `override fun onDownloadStarted(string: String) = Unit`

File: catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.utils
+
+import org.catrobat.catroid.scratchconverter.Client.ProjectDownloadCallback
+import org.catrobat.catroid.ui.recyclerview.fragment.MainMenuFragment
+import java.lang.ref.WeakReference
+
+object ProjectDownloadUtil : ProjectDownloadCallback {
+    private var fragment: WeakReference<MainMenuFragment>? = null
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/utils/ProjectDownloadUtil.kt:30]: please change it to `private lateinit var fragment: WeakReference<MainMenuFragment>` then in line 36, no need for `?` operator `fragment.get().setAndLoadCurrentProject()`

**Comment by wslany** [comment]
Thanks!
